plugin/mlsdbonlinelocator.cpp

```diff
@@ -259,11 +276,14 @@ QVariantMap MlsdbOnlineLocator::wifiAccessPointFields()
             // _nomap must NOT be used for privacy reasons."
             continue;
         }
-        QVariantMap wifiInfo;
-        wifiInfo["macAddress"] = service->bssid();
-        wifiInfo["frequency"] = service->frequency();
-        wifiInfo["signalStrength"] = service->strength();
-        wifiInfoList.append(wifiInfo);
+        
+        if(service->frequency() != 0) {
```

Comment:
if ()

Reply:
I was nitpicking over missing space between if and '('

Reply:
But yea, why does this filter out everything if frequency is not known? Feels some sort of a workaround. Should such service be even available here?

Reply:
It was revealed: that if the wifi network does not have data, then its parameters is zero. Filtration by frequency simplifies the code and in this case do not need to add a filter for all parameters

Reply:
If the item is completely bogus, IMO it shouldn't be advertised by Connman.

@monich ?

Reply:
Yes, frequency is not reported for wifi services without the network but I'm not 100% sure what happens when the network goes out of range, e.g. whether the value of the `"Frequency"` property would get stuck in `NetworkService` or not.

In our connman fork, services have `"Available"` property - I would rather use that to find the networks which are actually present in the air. When the network goes out of range, `"PropertyChanged"` signal is definitely issued for this property, so its value should follow the actual state.

Reply:
OfonoProvider sends the following data
[D] OfonoProvider::wifiAccessPointFields:149 -   wifiAccessPoints   macAddress =  "02:2d:0d:1a:22:14"  frequency 2457  signalStrength 33
[D] OfonoProvider::wifiAccessPointFields:149 -   wifiAccessPoints   macAddress =  ""  frequency 0  signalStrength 0

Reply:
Hmmm... wifi, mac address - what does it have to do with ofono?

Reply:
..with connman-qt5 /networkservice    m_wifiServices = m_networkManager->getServices("wifi");

Reply:
I'm using OS Sailfish version 2.1.4.2. Perhaps this has already been fixed